File: part1-exercises-for-rust/hardways/leetcode-maximum-depth-of-binary-tree/solution.c
struct TreeNode {
   int val;
    struct TreeNode *left;
    struct TreeNode *right;
};

int maxDepth(struct TreeNode* root){
    if(root==0) 
        return 0;
    int height = 1;
    if(root->left)
        height += maxDepth(root->left);
    if(root->right){
        int a =  maxDepth(root->right)+1;
        height = a > height? a:height;
    }
    return height;
}

File: part1-exercises-for-rust/hardways/leetcode-maximum-depth-of-binary-tree/solution.rs
use std::rc::Rc;
use std::cell::RefCell;
impl Solution {
    pub fn max_depth(root: Option<Rc<RefCell<TreeNode>>>) -> i32 {
        fn travel(root: &Option<Rc<RefCell<TreeNode>>>) -> i32{
            let mut d = 0;
            if let Some(ref n) = root {
                d = travel(&n.borrow().left);
                d = d.max( travel(&n.borrow().right));
                d = d+1;
            }else{
                d = 0;
            }
            d
        }
        travel(&root)
    }
}